compiler/src/iree/compiler/Codegen/Common/DecomposeConvolutionToLowerDimOps.cpp

```diff
@@ -16,6 +19,106 @@ namespace mlir::iree_compiler {
 
 namespace {
 
+static bool foldHDim(linalg::DepthwiseConv2DNhwcHwcOp convOp) {
+  Value kernel = convOp.getInputs().back();
+  Value output = convOp.getOutputs().front();
+
+  auto kernelType = dyn_cast<RankedTensorType>(kernel.getType());
+  auto outputType = dyn_cast<RankedTensorType>(output.getType());
+
+  auto kernelShape = kernelType.getShape();
+  auto outputShape = outputType.getShape();
+
+  int64_t khSize = kernelShape[0];
+  int64_t ohSize = outputShape[1];
+  bool removeH = (khSize == 1 && ohSize == 1);
+
+  return removeH;
+}
+
+/// Computes a "decomposed" lowering config attribute for a conv OP
+///
+/// This method complements the patterns to decompose 2D convolutions into 1D
+/// convs. Specifically, it will update the lowering config attached to a Conv
+/// Op in a way that matches the "decomposition" patterns.
+///
+/// At the moment only Depthwise HWC convolutions are supported.
+static FailureOr<IREE::Codegen::LoweringConfigAttr>
+computeNewLCA(ArrayRef<Operation *> computeOps, MLIRContext *context) {
+
+  // 0.1 Double-check that there's only one convolution Op.
+  // TODO: Make this hook work with multiple conv Ops
+  assert(llvm::count_if(computeOps,
+                        [](Operation *op) {
+                          return isa<linalg::ConvolutionOpInterface>(op);
+                        }) == 1 &&
+         "Exactly 1 Linalg Conv Op is expected");
+
+  // 0.2 ATM only 2D depthwise HWC convs are supported.
+  // TODO: Add support for other convs
+  bool seen2DDepthwiseConvHWC = false;
+  linalg::DepthwiseConv2DNhwcHwcOp convOp;
+  for (auto op : computeOps) {
+    if (isa<linalg::DepthwiseConv2DNhwcHwcOp>(op)) {
+      seen2DDepthwiseConvHWC = true;
+      convOp = cast<linalg::DepthwiseConv2DNhwcHwcOp>(op);
+      break;
+    }
+  }
+  if (!seen2DDepthwiseConvHWC)
+    return failure();
+
+  // 0.3. ATM only folding of the H dim is supported.
+  // TODO: Add support for cases where the W dim is folded.
+  if (!foldHDim(convOp))
+    return failure();
+
+  // 1. Get the current lowering config attached to the Conv Op.
+  FailureOr<IREE::Codegen::LoweringConfigAttr> loweringConfigAttr =
+      getLoweringConfig(computeOps);
+  if (failed(loweringConfigAttr))
+    return failure();
+
+  TilingConfig tc(loweringConfigAttr.value());
```

Comment:
`TilingConfig` is a class that currently only makes sense for CPU lowering pipelines, and LLVMGPU uses these patterns: https://github.com/openxla/iree/blob/a2476ceef53f6d4ec0f7a9cf36892647626d324b/compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp#L203

Because this pattern is only being applied to depthwise convs, and there are no tests on the GPU side for depthwise conv, there are no test failures. However given that these patterns are common, we need to find a way to separate out the target specific parts.

Vectorization has the same problem; TilingConfig being a CPU specific concept is used to interpret LoweringConfigAttrs for all backends in a common set of patterns. Can you add an option to this pass to allow disabling the lowering config logic here? I don't want to add a big blocker here given that, going by the tests, this isn't hitting any critical paths on other backends. cc @MaheshRavishankar

Reply:
It looks like this is unused to me (the `TilingConfig` is constructed, but never referenced in the function).

Reply:
I think it is fine because we drop the dimension for all the lists. The idea is not CPU specific. Let's remove TilingConfig since it is not used.